Products/CMFPlone/controlpanel/browser/actions.py

```diff
@@ -163,6 +165,14 @@ def set_position(self, value):
 
     position = property(get_position, set_position)
 
+    def get_modal(self):
+        return json.loads(self.context.modal)
+
+    def set_modal(self, value):
+        self.context.modal = json.dumps(value)
```

Comment:
@mauritsvanrees not sure if this loads/dumps is needed for the textinput here ... the JSON is validated in `plone.base.interfaces.controlpanel.IActionSchema`

Reply:
No, it is not needed. With this code, I actually get a validation error when I edit the Delete action and save it without changes: "Expecting property name enclosed in double quotes."
Ah, in the ZMI the value is:

    {"actionOptions": {"disableAjaxFormSubmit":true, "redirectOnResponse":true}}

With the current code in this PR, this shows up in the Plone control panel as:

    {'actionOptions': {'disableAjaxFormSubmit': True, 'redirectOnResponse': True}}

And this is no json, due to the single quotes and the Python-style booleans.
Funny. :-)

So just use `return self.context.modal` and `self.context.modal = value`. Then it works for me.
I have pushed a fix.